app/controllers/checkout_controller.rb

```diff
@@ -78,6 +78,18 @@ def confirm_order
 
     return true if redirect_to_payment_gateway
 
+    # Redeem VINE voucher
+    vine_voucher_redeemer = VineVoucherRedeemerService.new(order: @order)
+    if vine_voucher_redeemer.call == false
+      # rubocop:disable Rails/DeprecatedActiveModelErrorsMethods
+      flash[:error] = if vine_voucher_redeemer.errors.keys.include?(:redeeming_failed)
+                        vine_voucher_redeemer.errors[:redeeming_failed]
+                      else
+                        I18n.t('checkout.errors.voucher_redeeming_error')
+                      end
```

Comment:
Can't we rely on `vine_voucher_redeemer` to give us the error we need to know about? IE:
```suggestion
      flash[:error] = vine_voucher_redeemer.errors.first
```

Or if not, maybe we can use optional chaining like so:
```suggestion
      flash[:error] = vine_voucher_redeemer.errors&[:redeeming_failed] ||
                      I18n.t('checkout.errors.voucher_redeeming_error')